python/lsst/analysis/tools/tasks/propertyMapTractAnalysis.py

```diff
@@ -0,0 +1,179 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = [
+    "PropertyMapTractAnalysisConfig",
+    "PropertyMapTractAnalysisTask",
+]
+
+from typing import Mapping
+
+from lsst.daf.butler import DataCoordinate
+from lsst.pex.config import Config, ConfigDictField, Field, ListField
+from lsst.pipe.base import ButlerQuantumContext, InputQuantizedConnection, OutputQuantizedConnection
+from lsst.pipe.base import connectionTypes as ct
+from lsst.skymap import BaseSkyMap
+
+from ..interfaces import AnalysisBaseConfig, AnalysisBaseConnections, AnalysisPipelineTask
+
+
+class PropertyMapConfig(Config):
+    coaddName = Field(
+        dtype=str,
+        doc="Coadd name: typically one of deep or goodSeeing.",
+        default="deep",
+    )
+    operations = ListField(
+        dtype=str,
+        doc="List of operations whose corresponding maps should be retrieved.",
+        default=["min", "max", "mean", "weighted_mean", "sum"],
+    )
+
+
+class PropertyMapTractAnalysisConnections(
+    AnalysisBaseConnections,
+    dimensions=("skymap", "tract", "band"),
+    defaultTemplates={"outputName": "propertyMapTract"},
+):
+    healSparsePropertyMapsConfig = ct.Input(
+        doc="Configuration parameters for HealSparseInputMapTask in pipe_tasks.",
+        name="healSparsePropertyMaps_config",
+        storageClass="Config",
+        dimensions=(),
+    )
+
+    skymap = ct.Input(
+        doc="The skymap that covers the tract that the data is from.",
+        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+
+        operationNameLookup = {
+            "min": "Minimum",
+            "max": "Maximum",
+            "mean": "Mean",
+            "weighted_mean": "Weighted mean",
+            "sum": "Sum",
+        }
+
+        # Making connections for the maps that are configured to run.
+        for propertyName in config.properties:
+            coaddName = config.properties[propertyName].coaddName
+            for operationName in config.properties[propertyName].operations:
+                operationLongName = operationNameLookup[operationName]
+                name = f"{coaddName}Coadd_{propertyName}_map_{operationName}"
+                setattr(
```

Comment:
Does this actually work for the connections?  Because before one of Jim's refactoring it didn't.  But if it does now, that's great!